http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/PersistentDiskComponent.scala

```diff
@@ -174,12 +180,11 @@ object persistentDiskQuery {
       .filter(_.name === name)
       .filter(_.destroyedDate === dummyDate)
 
-  private[db] def findByNameQuery(cloudContext: CloudContext, name: DiskName) =
+  private[db] def findActiveByNameWorkspaceQuery(workspaceId: WorkspaceId, name: DiskName) =
     tableQuery
-      .filter(
-        _.cloudContext === cloudContext.asCloudContextDb
-      )
+      .filterOpt(Some(workspaceId))(_.workspaceId === _)
```

Comment:
Here, we assume uniqueness based on Name + workspaceId + destroyedDate. We should likely have a db constraint to enforce this so it does not allow insertion of resources that violate this assumption.

Reply:
the syntax can be found in multiple places in liquibase code ie.

`<addUniqueConstraint columnNames="googleProject, clusterName, destroyedDate" constraintName="IDX_CLUSTER_UNIQUE" tableName="CLUSTER"/>`